### PR TITLE
[GNUMake] Add Builder

### PR DESCRIPTION
**Pull request by Wimmerer**
First pass. Excluded Windows. Believe there is an issue related to http://gnu-make.2324884.n4.nabble.com/REGRESSION-cross-compile-4-3-fail-on-mingw-w64-7-0-GCC-9-2-td19810.html

### CODE REVIEW

**Comment by Wimmerer** [comment]
Ok I'll work on mingw tomorrow. Question though. Is it Make, make or gmake for the name here?

**Comment by Wimmerer** [comment]
Compiling make version 4.2 I seem to only be missing `glob.h`:
```cc -DHAVE_CONFIG_H -I. -I..  -I./include -I./subproc -I..   -g -O2 -MT compat/libw32_a-posixfcn.o -MD -MP -MF compat/.deps/libw32_a-posixfcn.Tpo -c -o compat/libw32_a-posixfcn.o `test -f 'compat/posixfcn.c' || echo './'`compat/posixfcn.c
In file included from pathstuff.c:17:0:```
`../makeint.h:453:18: fatal error: glob.h: No such file or directory`
`#include <glob.h>`

`glob.h` _is_ in ./workspace/srcdir/make-4.2/glob/glob.h and ./opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/glob.h
I assume it's wanting a mingw specific glob.h?

I'll investigate if msys2 does something different here.

**Comment by Wimmerer** [comment]
http://savannah.gnu.org/bugs/?48037

**Comment by Wimmerer** [comment]
Not totally happy with going around the makefile. It should work with Windows, but a straight up translation of the .bat file for mingw installation was recommended for other people attempting this, and it works without other changes.

File: M/GNUMake/bundled/patches/make-4.3_undef-HAVE_STRUCT_DIRENT_D_TYPE.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
++++ b/src/config.h.W32
+@@ -294,7 +294,7 @@ this program.  If not, see <http://www.gnu.org/licenses/>.  */
+ /* Define to 1 if `d_type' is a member of `struct dirent'. */
+ /* SV 57152: MinGW64 version of dirent doesn't support d_type. */
+ #ifndef __MINGW64__
+-# define HAVE_STRUCT_DIRENT_D_TYPE 1
++/* # define HAVE_STRUCT_DIRENT_D_TYPE 1 */
+ #endif
+ 
+ /* Define to 1 if you have the `isatty' function. */
```
Comment (giordano) [inline M/GNUMake/bundled/patches/make-4.3_undef-HAVE_STRUCT_DIRENT_D_TYPE.patch:13]:
```suggestion
 /* Define to 1 if you have the `isatty' function. */

```

**Review by giordano — CHANGES_REQUESTED** [review]
Some comments already given on Slack:

* the directory should be renamed to `G/GNUMake`
* a simple Makefile for Windows would be way more straightforward, maintainable and easy to read than the current manually written commands

File: G/GNUMake/bundled/Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+CPPFLAGS = -DWINDOWS32 -DHAVE_CONFIG_H -I./src -I./lib -I./src/w32/include -I./src/w32 -I./src/w32/subproc
+CFLAGS = -mthreads -Wall -std=gnu99 -gdwarf-2 -g3 -O2
+
+
+
+make: src/ar.o src/arscan.o src/commands.o src/default.o src/dir.o src/expand.o src/file.o src/function.o src/getopt.o src/getopt1.o src/guile.o src/hash.o src/implicit.o src/job.o src/load.o src/loadapi.o src/main.o src/misc.o src/output.o src/read.o src/remake.o src/remote-stub.o src/rule.o src/signame.o src/strcache.o src/variable.o src/version.o src/vpath.o src/w32/pathstuff.o src/w32/w32os.o src/w32/compat/posixfcn.o src/w32/subproc/sub_proc.o src/w32/subproc/misc.o src/w32/subproc/w32err.o lib/fnmatch.o lib/glob.o lib/getloadavg.o
+	cc $(CPPFLAGS) $(CFLAGS) -o $(bindir)/make$(exeext)  -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 $^ -Wl,--out-implib=$(prefix)/lib/libgnumake-1.dll.a
```
Comment (giordano) [inline G/GNUMake/bundled/Makefile:7]: I just realised with probably don't need to pass `CPPFLAGS` to link `make.exe`, let's try:
```suggestion
	cc $(CFLAGS) -o $(bindir)/make$(exeext)  -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -luuid -lodbc32 -lodbccp32 $^ -Wl,--out-implib=$(prefix)/lib/libgnumake-1.dll.a
```

**Review by giordano — APPROVED** [review]

**Pull request merged**